### PR TITLE
Fix some issues and support CI_MODE for captcha

### PR DESCRIPTION
**Pull request by nrbnlulu**
## Summary by Sourcery

This pull request introduces support for CI_MODE to bypass captcha validation during login, fixes issues related to JWT expiration and time formatting, enhances the user registration mutation, updates the CI configuration, and adds new tests and documentation.

- **New Features**:
    - Added support for CI_MODE to disable captcha validation during login, facilitating automated sign-in processes.
- **Bug Fixes**:
    - Fixed an issue where the JWT expiration check used the wrong time function, now using utc_now for consistency.
    - Corrected the JWT_TIME_FORMAT setting to use a valid strftime string.
- **Enhancements**:
    - Refactored the user registration mutation to improve readability and error handling.
    - Updated the Django version matrix in the CI configuration to include Django 5.0.7.
- **CI**:
    - Modified the CI workflow to exclude Python 3.8 and 3.9 for Django 5.0.7.
- **Documentation**:
    - Added RELEASE.md to document the release notes and resolved issues.
- **Tests**:
    - Added a new test to verify login functionality with CI_MODE enabled and invalid captcha.
    - Updated existing tests to align with the new CI_MODE feature and other changes.

### CODE REVIEW

**Comment by gqlauth** [comment]
### Release Notes
<details>
<summary>Just wanted to say</summary>

success

</details>
Release type: minor

resolve [#567](https://github.com/nrbnlulu/strawberry-django-auth/issues/567)
resolve [#564](https://github.com/nrbnlulu/strawberry-django-auth/issues/564)

Add support for CI mode when using captcha on login.
this should enable to automate the sign-in process by not validation the captcha fields.

File: gqlauth/jwt/types_.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ class TokenType:
     token: str = strawberry.field(description="The encoded payload, namely a token.")
 
     def is_expired(self):
-        return self.payload.exp < (datetime.utcnow())
+        return self.payload.exp < utc_now()
```
[the remark this answers is not in the document]
Comment (ilbertt) [reply gqlauth/jwt/types_.py:100]: @nrbnlulu this is exactly the bug that I'm facing when upgrading from v0.376.11 to v0.377.0. How can I fix it?

[the remark this answers is not in the document]
Comment (ilbertt) [reply gqlauth/jwt/types_.py:100]: @sourcery-ai I have `strawberry-django-auth==0.377.0` in my requirements and the error is:
```bash
.../site-packages/gqlauth/jwt/types_.py", line 100, in is_expired
    return self.payload.exp < utc_now()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: can't compare offset-naive and offset-aware datetimes
```

[the remark this answers is not in the document]
Comment (nrbnlulu) [reply gqlauth/jwt/types_.py:100]: @sourcery-ai when are you going to dispose us?

**Pull request merged**